draft-group-privacypass-k-check.md

```diff
@@ -62,23 +62,118 @@ The following terms are used throughout this document:
+The Mirror URI Template uses the Level 3 encoding defined {{Section 1.2 of RFC6570}} and contains
+one variables: "target", which is the percent-encoded URL of a HTTP resource to be mirrored.
+Example Mirror URI Templates are shown below.
+
+~~~
+https://mirror.example/mirror{?target}
+https://mirror.example/{target}
+~~~
+
+The Mirror URI Template MUST contain the "target" variable exactly once. The variable
+MUST be within the path or query components of the URI.
+
+Clients send requests to mirror resources after being configured with their corresponding Mirror
+URI Template. Clients MUST ignore configurations that do not conform to this template.
+
+Upon receipt of a mirror request, mirrors validate the incoming request. If the request is invalid
+or malformed, e.g., the "target" parameter is not a correctly encoded URL, the mirror aborts
+and returns a a 4xx (Client Error) to the client. Otherwise, if the request is valid, the mirror
+checks to see if it has a cached version of the resource identified by the target URL. If so, this
+response is returned to the client.
+
 Otherwise, mirrors send a GET request to the target resource URL. If this request fails,
-the mirror returns an error to the client. Otherwise, the response to a mirror request is
-the content that was contained in the target resource. With the exception of Cache-Control
-headers, the mirror response does not replicate any of the headers that may have been attached
-to the HTTP message carrying the target resource.
+the mirror returns a 4xx error to the client. Otherwise, the response to a mirror request is
+the content that was contained in the target resource. With the exception of Content-Type,
+Content-Length, and Cache-Control headers, the mirror response does not replicate any of the
+headers that may have been attached to the HTTP message carrying the target resource.
 
-[[OPEN ISSUE: Should there be some mandatory or RECOMMENDED minimum validity window of resources?]]
+Mirrors cache this content, indexed by the target URL, for future mirror requests. Mirrors
+purge this cache when the response is no longer valid according to the Cache-Control headers.
 
 We refer to the set of clients that interact with a mirror as mirror clients. We refer to
 the validity window of the mirror response as the period of time determined by the Cache-Control
 headers as the response.
 
+[[OPEN ISSUE: Should there be some mandatory or RECOMMENDED minimum validity window of resources?]]
+
+## Mirror Request and Respnose Example
+
+The following example shows two mirror request and response examples. The first one yields a mirror
+cache miss and the second one yields a mirror cache hit. The Mirror URI Template is
+"https://mirror.example/mirror{?target}", and the target URL is
+"https://issuer.example/.well-known/private-token-issuer-directory".
+
+The first client request to the mirror might be the following.
```

Comment:
Should this take into account the Accept header from the client?

Reply:
As in, should the mirror, uh, mirror the Accept header from the client in its request to the target? If so, that seems like a good idea.

Reply:
Yeah, since different clients may expect different things or accept different variants/content-types. Though this might also complicate the caching logic if different Accept values result in different resource responses from the target.

Reply:
I had to check the spec, but it turns out that Accept doesn't complicate caching as much as the Vary header, so I left this in.